service/s3/api_op_ListParts.go

```diff
@@ -334,8 +334,10 @@ func (p *ListPartsPaginator) NextPage(ctx context.Context, optFns ...func(*Optio
 	p.firstPage = false
 
 	prevToken := p.nextToken
-	p.nextToken = result.NextPartNumberMarker
+	if result.IsTruncated {
+		p.nextToken = result.NextPartNumberMarker
 
+	}
```

Comment:
I am pretty sure we also need to set `p.nextToken` to `nil`, otherwise won't it just retain the token of the last page? And then not stop just as before.

The current fix will only work if there is only a single page returned.